src/base/http/connection.cpp

```diff
@@ -97,7 +98,7 @@ void Connection::read()
 
         case RequestParser::ParseStatus::OK:
         {
-                const Environment env {m_socket->localAddress(), m_socket->localPort(), m_socket->peerAddress(), m_socket->peerPort()};
+                const Environment env {m_socket->localAddress(), m_socket->localPort(), resolvePeerAddress(result.request), m_socket->peerPort()};
```

Comment:
My main comment to the implementation last time was that it is at the wrong level.
Here (namely, in the Connection class), only the request is received and parsed. The processing should be done at a higher level (in http request handler instance).
`Environment` is intended to provide real data (related to TCP connection).

Reply:
IF I understand correctly, it should be done in https://github.com/qbittorrent/qBittorrent/blob/40f6201509bf40097e7d3caa70d0e6c5494c0f3c/src/webui/webapplication.cpp#L469 ?

Reply:
>If I understood correctly, it should be done in qBittorrent/src/webui/webapplication.cpp?

If you intend to cover only the WebAPI/WebUI, then Yes.

FYI, there is another implementation of RequestHandler interface. It's embedded Tracker. But I don't remember it being mentioned in relation with the reverse proxy. Maybe it's not even affected by it.